app/models/offer.rb

```diff
@@ -207,6 +208,43 @@ def nondraft_states
     def donor_states
       valid_states - ["draft"]
     end
+
+    def priority_and_non_priority_offers_count_for(self_reviewer: false)
+      non_priority_active_offers_count(self_reviewer).merge(
+        priority_active_offers_count(self_reviewer)
+      )
+    end
+
+    def non_priority_active_offers_count(self_reviewer)
+      all_offers_count = active_offers_count_as_per_priority_and_state(self_reviewer)
+      total = all_offers_count.values.reduce(:+)
+      all_offers_count["active_offers_total_count"] = total
+      all_offers_count = prepend_keys_with_reviewer(all_offers_count) if self_reviewer
+      all_offers_count
+    end
+
+    def priority_active_offers_count(self_reviewer)
+      all_offers_count = active_offers_count_as_per_priority_and_state(self_reviewer, is_priority: true)
+      all_offers_count = prepend_keys_with_reviewer(all_offers_count) if self_reviewer
+      all_offers_count
+    end
+
+    def prepend_keys_with_reviewer(offers_count_hash)
+      offers_count_hash.transform_keys { |key| "reviewer_".concat(key) }
+    end
+
+    def active_offers_count_as_per_priority_and_state(self_reviewer, is_priority: false)
+      offers = filter(state_names: ACTIVE_OFFERS, priority: is_priority, self_reviewer: self_reviewer).group_by(&:state)
+      if is_priority
+        offers_count_per_state(offers).transform_keys { |key| "priority_".concat(key) }
+      else
+        offers_count_per_state(offers)
+      end
+    end
+
+    def offers_count_per_state(offers)
+      offers.each { |key, value|  offers[key] = value.count }
```

Comment:
maybe rename the `offers` variable, it made me think it was an array of offers (actually a hash of states)